src/Lean/PrettyPrinter/Delaborator/Builtins.lean

```diff
@@ -455,13 +457,16 @@ private def shouldGroupWithNext : DelabM Bool := do
   | Expr.lam _ _     e'@(Expr.lam _ _ _ _) _     => go e'
   | Expr.forallE _ _ e'@(Expr.forallE _ _ _ _) _ => go e'
   | _ => pure false
+where
+  getPPBinderTypes (e : Expr) :=
+    if e.isForall then getPPPiBinderTypes else getPPFunBinderTypes
 
 private partial def delabBinders (delabGroup : Array Syntax → Syntax → Delab) : optParam (Array Syntax) #[] → Delab
   -- Accumulate names (`Syntax.ident`s with position information) of the current, unfinished
   -- binder group `(d e ...)` as determined by `shouldGroupWithNext`. We cannot do grouping
   -- inside-out, on the Syntax level, because it depends on comparing the Expr binder types.
   | curNames => do
-    if (← shouldGroupWithNext) then
+    if ← shouldGroupWithNext then
```

Comment:
Ah, I know that I've use the latter style in the delaborator, but Leo prefers the one with parentheses. Just something to keep in mind for the future. We should fix all existing occurrences automatically at some point...